### PR TITLE
Add 64-bit support and fix ADODB field tests

### PR DESCRIPTION
**Pull request by dionrhys**
This PR adds 64-bit support by using the correct size of the native `Variant` structure in IDispatchAccess.cs.

The conditional to determine if running on 32-bit or not (`IntPtr.Size == 4`) may seem arbitrary, but it matches exactly the condition Microsoft uses in the [reference source](https://referencesource.microsoft.com/#mscorlib/system/Runtime/InteropServices/Variant.cs) linked in the code.

This PR also fixes the unit tests so that they work in VS 2019 (albeit with many analyser warnings cause of the state of the tests), and fixes an issue with the ADODB field tests where it was using an incorrect equality check. 4 tests fail - these were failing before the changes here.

Finally .gitignore is added (since this repo is now using Git not Mercurial), and the version bumped to 1.9.4.

_We've been running this code since 2019 so it's had a lot of time working in production._

### CODE REVIEW

**Comment by ProductiveRage** [comment]
Looks good to me! Have only been able to examine the changes on my phone, not build and test as my internet dropped out. Is it worth addinf the analyser package if it throws up so many warnings or do you think that there's value to those warnings being addressed?

**Comment by ProductiveRage** [comment]
Er.. did I close this instead of merge?? Let me try again.

**Comment by dionrhys** [comment]
> 
> 
> Looks good to me! Have only been able to examine the changes on my phone, not build and test as my internet dropped out. Is it worth addinf the analyser package if it throws up so many warnings or do you think that there's value to those warnings being addressed?

Most of the analyser warnings are verbose (e.g. `xUnit1014:MemberData should use nameof operator for member name` and `xUnit1026:Theory methods should use all of their parameters`), but others do flag up correctness issues (e.g. "The literal or constant value ... should be passed as the 'expected' argument [instead of the 'actual' argument]").

The analyzer came automatically with the xUnit upgrade but I guess it could be uninstalled manually (though it'll reappear at the next package update).

It would be nice to have the test warnings addressed at some point, _by monkeys_.

Alternatively we could reduce the severity of `xUnit1026:Theory methods should use all of their parameters` and `xUnit1014:MemberData should use nameof operator for member name` to be suggestions or silent instead. This then leaves 5 warnings to fix (and excluding the test failures that already occur due to TODOs).
